### PR TITLE
[ANCHOR-518] Add @LockAndMockStatic and @LockStatic annotations

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

Add @LockAndMockStatic and @LockStatic annotations. Instead of 
```kotlin
class MyTest {
  @LockAndMockStatic([NetUtil::class]) 
  fun myTest(){
    lockAndMockStatic() {
      // my tests here
   }
  }
}
```
the test method can be annotated with `@LockAndMockStatic`

```kotlin
@ExtendWith(LockAndMockTest::class) 
class MyTest {
  @LockAndMockStatic([NetUtil::class]) 
  fun myTest(){
  }
}
```

### Context

To simplify the `lockAndMock` function.

### Testing

- `./gradlew test`

### CODE REVIEW

File: core/src/test/kotlin/org/stellar/anchor/auth/AuthHelperTest.kt [hunk 1]
```diff
@@ -22,8 +22,8 @@ class AuthHelperTest {
   @ParameterizedTest
   @EnumSource(AuthType::class)
   fun `test AuthHeader creation based on the AuthType`(authType: AuthType) {
-    val calendarSingleton = mockk<Calendar>(relaxed = true)
     lockAndMockStatic(Calendar::class) {
```
Comment (philipliu) [inline core/src/test/kotlin/org/stellar/anchor/auth/AuthHelperTest.kt:25]: Should this be replaced by the annotation?

**Review by philipliu — APPROVED** [review]
LGTM

File: core/src/test/kotlin/org/stellar/anchor/auth/AuthHelperTest.kt (hunk 1 above)
Comment (lijamie98) [reply core/src/test/kotlin/org/stellar/anchor/auth/AuthHelperTest.kt:25]: Probably not. I think ```val calendarSingleton = mockk<Calendar>(relaxed = true)``` should happen before the mock.

Comment (philipliu) [reply core/src/test/kotlin/org/stellar/anchor/auth/AuthHelperTest.kt:25]: Didn't notice that 👍

**Pull request merged**